### ISSUE TITLE
Dr.VAE: Drug Response Variational Autoencoder

### ISSUE THREAD

**Issue by agitter** [greenelab/deep-review#556]
https://arxiv.org/abs/1706.08203

> We present two deep generative models based on Variational Autoencoders to improve the accuracy of drug response prediction. Our models, Perturbation Variational Autoencoder and its semi-supervised extension, Drug Response Variational Autoencoder (Dr.VAE), learn latent representation of the underlying gene states before and after drug application that depend on: (i) drug-induced biological change of each gene and (ii) overall treatment response outcome. Our VAE-based models outperform the current published benchmarks in the field by anywhere from 3 to 11% AUROC and 2 to 30% AUPR. In addition, we found that better reconstruction accuracy does not necessarily lead to improvement in classification accuracy and that jointly trained models perform better than models that minimize reconstruction error independently.

Bonus points for the title.

**Comment by gwaygenomics** [comment]
## General Summary

The authors train two variational autoencoders (VAEs) to model cancer cell lines reaction and susceptibility to drug treatments. One of the VAEs includes a semi-supervised component that predicts treatment response. The paper presents the models and data effectively while adding several valuable computational evaluations, but the paper lacks biological interpretation. I am also not entirely sold that their reported results support one of their conclusions that doing well on reconstruction does not lead to classification improvements (although I tend to agree that reconstruction and biological meaning are not monotonic). There is no available source code as far as I can see.

## Biological Aspects

The models are trained using ~600 cell line gene expression profiles using measurements for 903 landmark genes. The cell lines are measured before and after treatment by 19 drugs. Models were trained independently for each drug. 

## Computational Aspects

Two VAE models train with a combination of SGD and Inverse Autoregressive Flow. Both learn two distinct but linearly-related latent representations for pre and post drug perturbation. The first model, `Perturbation VAE` learns to encode the response to drug treatment, while the related `Drug Response VAE` or "Dr.VAE" is the semi-supervised extension that predicts drug mediated cell killing.

## Evaluation

1. Compare drug response predictions for Dr.VAE against a generic semi-supervised VAE, logistic regression, random forest, and SVMs. AUROC + AUPR
    - Dr.VAE convincingly outperforms all other models
    - Use latent embedding features in this classification task - PertVAE vs. 100 component PCA linear regression
2. Compare gene expression reconstruction "quality" - rank correlation between reconstruction mean and original gene expression (same number of dimensions pertVAE vs. PCA reconstruction)
    - Conclusion is VAE does better with smaller latent spaces, then "overfits" as compared to PCA in more latent dimensions.
    - Mean correlations don't capture reconstruction quality _coverage_, which I think could have been a more compelling comparison.
3. Compare mean latent space rank correlation (observed and predicted) vs. input gene expression rank correlation (observed and predicted)
    - Conclusion is that if the correlation is higher for the latent features, then it the method learns something more than just reconstruction
    - Cool idea as the predictions should have higher correlation if more than just structure is learned. However, mean correlations lose resolution again.
